apps/etl-func/src/adapters/cosmos/event-collector.ts

```diff
@@ -1,63 +1,56 @@
-import {
-  EventCollector,
-  EventsSummary,
-  eventsSummarySchema,
-} from "@/domain/event.js";
+import { EventCollector, EventsSummary } from "@/domain/event.js";
 import { TelemetryEventName, TelemetryService } from "@/domain/telemetry.js";
 import { Container, ItemResponse } from "@azure/cosmos";
 
 /**
- * Adapter for a cosmos weekly event collector.
- * The goal is to store the number of events sent to the event hub
- *  into a container with year as partition key and week number as model id.
+ * Adapter for a cosmos event collector.
+ * The goal is to store the number of events
+ * into a container using the year as partition key.
  * */
-export class CosmosWeeklyEventCollector<T> implements EventCollector<T> {
+export class CosmosSummaryCollector<T extends { id: string }>
+  implements EventCollector<T>
+{
   #container: Container;
   #telemetry: TelemetryService;
 
   /**
    * @param container The name of the container where to store weekly events.
+   * @param telemetry The telemetry service to use
    * */
   constructor(container: Container, telemetry: TelemetryService) {
     this.#container = container;
     this.#telemetry = telemetry;
   }
 
-  private createSummary(events: T[]): EventsSummary {
+  /**
+   * Accept a batch of events and return the summary.
+   * */
+  private createSummary(events: [T, ...T[]]): EventsSummary {
     return {
       count: events.length,
-      id: this.getCurrentModelId(),
+      id: this.getBatchId(events),
       year: this.getCurrentPartitionKey(),
     };
   }
 
-  private getCurrentModelId(): string {
-    const today = new Date();
-    const startOfYear = new Date(today.getFullYear(), 0, 1);
-    const daysSinceStartOfYear = Math.floor(
-      (today.getTime() - startOfYear.getTime()) / 86_400_000,
-    );
-    const weekNumber = Math.ceil(
-      (daysSinceStartOfYear + startOfYear.getDay() + 1) / 7,
-    );
-    return `${today.getFullYear()}-W${weekNumber.toString().padStart(2, "0")}`;
+  /**
+   * Returns the id of the first event in the batch.
+   *
+   * @param events Accept a non empty array
+   *
+   * @returns The id of the first event
+   * */
+  private getBatchId(events: [T, ...T[]]): string {
+    return events[0].id;
   }
```

Comment:
I would do this check at runtime (maybe using the `get` method of `Array`)

Reply:
I added inside the `collect` a runtime check in order to skip empty batches, we don't want to throw here because we would reprocess the entire batch [c230c9d](https://github.com/pagopa/io-messages/pull/130/commits/c230c9d392e911286bd57d5020b2ff2b2bfbcef4)